flask_authz/casbin_enforcer.py

```diff
@@ -16,18 +16,26 @@ class CasbinEnforcer:
 
     e = None
 
-    def __init__(self, app, adapter, watcher=None):
+    def __init__(self, app=None, adapter=None, watcher=None):
         """
         Args:
             app (object): Flask App object to get Casbin Model
             adapter (object): Casbin Adapter
         """
-        self.app = app
+        self.app = None
```

Comment:
To quote the example in the flask docs it looks like `self.app = app` and not None:
```
        self.app = app
        if app is not None:
            self.init_app(app)
```

Reply:
I think we should only make a declaration here. The initialization operation relying on app is moved to init_ app. thanks

Reply:
This makes having `app=None` kwarg pointless if `self.app=None`.  
I would probably suggest the following change:
```
self.app = app
if self.app is not None:
    self.init_app(self.app)
```
This seems like a more straight forward implementation.

Reply:
This seems to be a more popular and cleaner implementation.

    self.app = app
    if app is not None:
        self.init_app(app)